src/runtime-rs/crates/runtimes/virt_container/src/container_manager/process.rs

```diff
@@ -218,14 +258,18 @@ impl Process {
     }
 
     pub async fn close_io(&mut self, agent: Arc<dyn Agent>) {
-        self.wg_stdin.wait().await;
+        if let Some(_) = self.passfd_io.take() {
+            // do nothing
+        } else {
+            self.wg_stdin.wait().await;
```

Comment:
Do we need a similar logic in the agent in the passfd case? I mean, do we need to always make sure that we drain the stdin steam before closing? Likely we have to handle it in dragonball's hvsock implementation.

Reply:
Oh yes, I agree, that's a problem. 

Now I am wondering if contained will really close the stdin fifo before or after sending the `CloseIORequest` rpc? 

If so, the guest af_vsock stream will finally be closed by hvsock implementation, then we just need to delay the closing after `stdin_stream` is drained in the agent.